packages/new-deploy/scripts/verify.ts

```diff
@@ -0,0 +1,245 @@
+// --chain-id
+// --compiler-version
+// --num-of-optimizations
+
+// address
+// path/to/Contracts.sol:ContractName
+// etherscan-key
+
+import { writeFile, readFile } from 'fs';
+import { exec } from 'child_process';
+import { promisify } from 'util';
+import { Verification } from '../src';
+import * as dotenv from 'dotenv';
+import { NomadContext } from '@nomad-xyz/sdk';
+dotenv.config();
+
+type VerificationOutput = Verification & {
+  GUID?: string;
+  verifyCommand?: string;
+  verified?: boolean;
+};
+
+const execAsync = promisify(exec);
+const readFileAsync = promisify(readFile);
+const writeFileAsync = promisify(writeFile);
+
+/// async sleep function
+function delay(ms: number): Promise<void> {
+  return new Promise((resolve) => setTimeout(resolve, ms));
+}
+
+const etherscanKey = process.env.ETHERSCAN_KEY;
+
+/// Extract the etherscan GUID from the `forge verify-contract` stdout
+export function extractGuidFromFoundry(res: {
+  stdout: string;
+  stderr: string;
+}): string | undefined {
+  if (res.stderr.indexOf('already verified') !== -1) return;
+  const bits = res.stdout.split('`');
+  if (bits.length < 4)
```

Comment:
ideally could just add some comments 
```
// we expect there to be 4 arguments
// the GUID is the 4th argument
```

Reply:
I'll leave a comment saying that it's lazy and brittle lol. next PR.